hyprgrass build failure

Using Hyprland 0.51.1
after running `hyprpm -v add https://github.com/horriblename/hyprgrass` build fails with this message:
```
...
✖ Plugin hyprgrass failed to build.
  This likely means that the plugin is either outdated, not yet available for your version, or broken.
  If you are on -git, update first
  Try re-running with -v to see more verbose output.
...
```

Before that, there is this message:
```
[12/15] Compiling C++ object src/libhyprgrass.so.p/TouchVisualizer.cpp.o
FAILED: src/libhyprgrass.so.p/TouchVisualizer.cpp.o 
c++ -Isrc/libhyprgrass.so.p -Isrc -I../src -Isubprojects/wf-touch -I../subprojects/wf-touch -I/usr/include -I/usr/include/pixman-1 -I/usr/include/libdrm -I/var/cache/hyprpm/ejnar/headersRoot/include -I/var/cache/hyprpm/ejnar/headersRoot/include/hyprland/protocols -I/var/cache/hyprpm/ejnar/headersRoot/include/hyprland -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O3 -std=c++23 -fPIC -DWLR_USE_UNSTABLE -MD -MQ src/libhyprgrass.so.p/TouchVisualizer.cpp.o -MF src/libhyprgrass.so.p/TouchVisualizer.cpp.o.d -o src/libhyprgrass.so.p/TouchVisualizer.cpp.o -c ../src/TouchVisualizer.cpp
../src/TouchVisualizer.cpp: In member function ‘void Visualizer::onRender()’:
../src/TouchVisualizer.cpp:57:37: error: cannot convert ‘<brace-enclosed initializer list>’ to ‘float’
   57 |         g_pHyprOpenGL->renderTexture(this->texture, dmg, { .a = 1.f, .round = 0, .discardActive = true });
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /var/cache/hyprpm/ejnar/headersRoot/include/hyprland/src/render/Renderer.hpp:7,
                 from ../src/TouchVisualizer.cpp:2:
/var/cache/hyprpm/ejnar/headersRoot/include/hyprland/src/render/OpenGL.hpp:187:57: note: initializing argument 3 of ‘void CHyprOpenGLImpl::renderTexture(SP<CTexture>, const Hyprutils::Math::CBox&, float, int, float, bool, bool, GLenum, GLenum)’
  187 |     void renderTexture(SP<CTexture>, const CBox&, float a, int round = 0, float roundingPower = 2.0f, bool discardActive = false, bool allowCustomUV = false,
      |                                                   ~~~~~~^
```

--- a/hyprpm.toml
+++ b/hyprpm.toml
@@ -37,6 +37,8 @@ commit_pins = [
     ["997fefbc1113323ed2bf5d782bdafc0d17532647", "40d9cc4d9529e1e81305bc2681082bdb45118ce9"], # texture class members refactor
     ["9958d297641b5c84dcff93f9039d80a5ad37ab00", "51da8d137e042c117fe46a5c019ec38c0de0342a"], # v0.49.0
     ["a907ecd4ff736a3a09410532b405a437eb48033c", "debbc2a4f9a4092856737d24eb01802e0e914e76"], # renderTexture changes
+    ["c4a4c341568944bd4fb9cd503558b2de602c0213", "7dafd92afedda246a9c1c4187784c3fc1d6261e5"], # v0.50.0
+    ["4e242d086e20b32951fdc0ebcbfb4d41b5be8dcc", "7dafd92afedda246a9c1c4187784c3fc1d6261e5"], # v0.50.1
     ## DO NOT EDIT THIS LINE: for auto pin script ##
 ]
 

--- a/scripts/ci/test-pin.sh
+++ b/scripts/ci/test-pin.sh
@@ -23,7 +23,19 @@ hyprgrassRev=${2:-main}
 hyprlandCommit="$(git ls-remote https://github.com/hyprwm/Hyprland.git "${hyprlandRev}" | cut -f 1)"
 hyprgrassCommit="$(git rev-parse "${hyprgrassRev}")"
 
-nix build --no-link "git+file://$(pwd)?rev=${hyprgrassCommit}&shallow=1#hyprgrassWithTests" \
-	--override-input hyprland "github:hyprwm/Hyprland/${hyprlandRev}" \
+hlUrl="github:hyprwm/Hyprland/${hyprlandRev}" 
+
+worktreeDir="pin-build"
+
+git worktree add "$worktreeDir" "$hyprgrassCommit"
+pushd pin-build
+
+# NOTE: nix build --override-input does not override input of inputs, i.e. hyprland/aquamarine
+# will not be updated, hence we need to run `flake update` :<
+nix flake update --override-input hyprland "$hlUrl"
+nix build --no-link '.#hyprgrassWithTests'
+
+popd
+git worktree remove "$worktreeDir"
 
 echo "[\"${hyprlandCommit}\", \"${hyprgrassCommit}\"], # ${hyprlandRev}"

